File: web-forms/src/store/evaluation/seguranca.js
import { defineStore } from 'pinia'
import { ref } from 'vue';

const alergiaBase = {
    tipo_alergia: null,
    quais_alergias: null,
    precaucao: null,
    cuidados_paliativos: null
}

export const segurancaModel = {
    alergias: [],
}

export const useSegurancaStore = defineStore('seguranca', () => {
    const seguranca = ref(JSON.parse(JSON.stringify(segurancaModel)))

    function adicionarAlergia() {
        seguranca.value.alergias.push({ ...alergiaBase })
    }

    function removerAlergia(index) {
        if (seguranca.value.alergias.length > 0) {
            seguranca.value.alergias.splice(index, 1)
        }
    }

    function reset() {
        seguranca.value = JSON.parse(JSON.stringify(segurancaModel))
    }

    return {
        seguranca,
        adicionarAlergia,
        removerAlergia,
        reset
    }
}, {
    persist: true
})

File: backend/internal/avaliation/domain/seguranca.go
package domain

type Seguranca struct {
	Alergias []Alergia `json:"alergias"`
}

type Alergia struct {
	TipoAlergia        SelectField `json:"tipo_alergia"` // FIXED: Object in JSON
	QuaisAlergias      string      `json:"quais_alergias"`
	Precaucao          SelectField `json:"precaucao"` // FIXED: Object in JSON
	CuidadosPaliativos string      `json:"cuidados_paliativos"`
}
